libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -93,8 +93,10 @@ void AP_AHRS_NavEKF::update(void)
     update_SITL();
 #endif
 
+#if AP_MODULE_SUPPORTED
     // call AHRS_update hook if any
     AP_Module::call_hook_AHRS_update(*this);
+#endif
```

Comment:
instead of doing this and the addition of AP_MODULE_SUPPORTED in the header of AP_Modules.h, I think it would be better to just ifdef inside the function. The compiler is pretty good to eliminate calls to empty functions and this way we don't spread the AP_MODULE_SUPPORTED ifdefs